### PR TITLE
ElasticSearch: Remember connection type and don't use cluster name

### LINKED ISSUE #1311: ElasticSearch datastore dialog requires cluster name even for REST connections
When connecting to an ElasticSearch datastore via the REST protocol, the dialog still requires me to specify cluster name, although this is to my knowledge not needed or even used when connecting via REST. But I have to specify it or else the "Register datastore" button is disabled.

image

### LINKED ISSUE #1331: ElasticSearchDatastore forgets REST setting
The ElasticSearchDatastores does not store the client type. That only affects REST clients, as loading has a backup way to detech the datastore type by looking if a port has been set (cluster connections does not use ports.

### PR DESCRIPTION
**Pull request by LosD**
This will store client connection types (none was remembered, but only REST was affected). Will also disable cluster name for REST clients, as it is not used.

 Fixes #1331
 Fixes #1311

### CODE REVIEW

File: desktop/ui/src/main/java/org/datacleaner/windows/ElasticSearchDatastoreDialog.java [hunk 1]
```diff
@@ -82,7 +80,6 @@ public ElasticSearchDatastoreDialog(WindowContext windowContext, MutableDatastor
         setSaveButtonEnabled(false);
 
         _clientTypeComboBox = new JComboBox<>(ClientType.values());
-        _clientTypeComboBox.setSelectedItem(DEFAULT_CLIENT_TYPE);
```
Comment (arjansh) [inline desktop/ui/src/main/java/org/datacleaner/windows/ElasticSearchDatastoreDialog.java]: With the removal of this line the private constant DEFAULT_CLIENT_TYPE is no longer used.

Comment (LosD) [inline desktop/ui/src/main/java/org/datacleaner/windows/ElasticSearchDatastoreDialog.java]: That's actually more of a failing in line 223, where I used ClientType.REST instead.

**Comment by arjansh** [comment]
Outside of my small inline comment it looks good to me.

**Comment by arjansh** [comment]
LGTM!

**Pull request merged**